src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security;
+
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.CodeAnalysis.BinaryParsers.ProgramDatabase;
+using Microsoft.CodeAnalysis.Sarif;
+
+namespace Microsoft.CodeAnalysis.IL.Sdk
+{
+    public class CompilerDataLogger : ICompilerDataLogger
+    {
+        private const string CompilerEventName = "CompilerInformation";
+
+        private readonly bool appInsightsRegistered;
+
+        private readonly string sha256;
+        private readonly string localPath;
+        private readonly string repositoryUri;
+        private readonly string pipelineName;
+
+        private static TelemetryClient s_telemetryClient;
+        private static TelemetryConfiguration s_telemetryConfiguration;
+
+        public CompilerDataLogger(IAnalysisContext analysisContext, string repositoryUri, string pipelineName)
+        {
+            try
+            {
+                string appInsightsKey = Environment.GetEnvironmentVariable("BinskimAppInsightsKey");
+                if (!string.IsNullOrEmpty(appInsightsKey) && Guid.TryParse(appInsightsKey, out _))
+                {
+                    Initialize(appInsightsKey);
+                    appInsightsRegistered = true;
+                }
+            }
+            catch (SecurityException)
+            {
+                // User does not have access to retrieve information from environment variables.
+            }
+
+            this.pipelineName = pipelineName;
+            this.repositoryUri = repositoryUri;
+            this.sha256 = analysisContext?.Hashes?.Sha256;
+            this.localPath = analysisContext?.TargetUri?.LocalPath;
+        }
+
+        public static void Initialize(string instrumentationKey)
+        {
+            s_telemetryConfiguration = new TelemetryConfiguration(instrumentationKey);
+            s_telemetryClient = new TelemetryClient(s_telemetryConfiguration);
+        }
+
+        public static void Flush()
+        {
+            s_telemetryClient?.Flush();
+        }
+
+        public void PrintHeader()
+        {
+            if (!appInsightsRegistered)
+            {
+                Console.WriteLine("RepositoryUri,PipelineName,Target,Compiler Name,Compiler BackEnd Version,Compiler FrontEnd Version,Language,Dialect,Module Name,Module Library,Hash,Error");
+            }
+        }
+
+        public void Write(string compilerData, ObjectModuleDetails omDetails)
+        {
+            string name = omDetails?.Name?.Replace(",", "_");
```

Comment:
>name [](http://example.com/codeflow?start=19&length=4)

are we really transforming data that previously existed in the PDB into some other normalized form? 

this doesn't seem like a good idea, it seems better to preserve data as it actually exists in the binary.

can you explain the decision? certainly this code needs to be commented if we retain what you've done.

Reply:
we were creating a csv and we were using ',' as separator.
with that, I saw cases where the name contained ',' breaking the csv.

this is why i changed from ',' to '_'

Reply:
typically, what you would do to workaround this would be to encapsulate the data in double quotes. is that an option here? 

we really should not be modifying the compiler generated data here. how will you convert it back if you need to unify it again with actual PDB data?